ai_ref_kits/conversational_ai_chatbot/README.md

```diff
@@ -0,0 +1,175 @@
+
+To use the model, authenticate using the Hugging Face CLI:
+
+```shell
+huggingface-cli login
+```
+When prompted to add the token as a git credential, respond with 'n'. This step ensures that you are logged into the Hugging Face API and ready to download the model.
+
+Now, you're ready to download and optimize the models required to run the application.
+
+## Model Conversion and Optimization
+
+_NOTE: This reference kit requires much bandwidth and disk space (>8GB) for downloading models. Also, the conversion may take much time (>2h) and need much memory (>32GB)) when running for the first time as the models used here are huge. After the first run, the subsequent runs will be done much faster._
+
+The application uses three separate models for its operation, each requiring conversion and optimization for use with OpenVINO™. Follow the order below to convert and optimize each model:
+
+1. Automated Speech Recognition Distil-Whisper Conversion:
+```shell
+python convert_and_optimize_asr.py --asr_model_type distil-whisper-large-v2 --precision int8
+```
+This script will convert and optimize the automatic speech recognition (ASR) model performing weights quantization.
+
+2. Chat LLama Conversion:
+```shell
+python convert_and_optimize_chat.py --chat_model_type llama3.1-8B --embedding_model_type bge-large --precision int4
+```
+This script will handle the conversion and optimization of:
+
+- The chat model (llama3.1-8B) with int4 precision.
+- The embedding model (bge-large) with FP32 precision.
+
+After running the conversion scripts, you can run app.py to launch the application.
+
+## Running the Application (Gradio Interface)
+
+_NOTE: This application requires much memory (>16GB) as the models used here are huge (especially chatbot). It may also be slow for less powerful devices._
+
+Execute the `app.py` script with the following command, including all necessary model directory arguments:
+```shell
+python app.py --asr_model_dir path/to/asr_model --chat_model_dir path/to/chat_model  --embedding_model path/to/embedding_model --public
+```
+Replace `path/to/asr_model` , `path/to/chat_model` ,`path/to/embedding_model` and other paths  with actual paths to your respective models. Add ``--public` to make it publicly accessible.
+
+### Accessing the Web Interface
+After running the script, Gradio will provide a local URL, typically `http://127.0.0.1:XXXX`, which you can open in your web browser to start interacting with the assistant. If you configured the application to be accessible publicly, Gradio will also provide a public URL.
+
+Trying Out the Application
+1. Navigate to the provided Gradio URL in your web browser.
+2. You will see the Gradio interface with options to input voice.
+3. To interact using voice:
+    - Click on the microphone icon and speak your query.
+    - Wait for the assistant to process your speech and respond.
+4. The assistant will respond to your query in text and audio form.
```

Comment:
Add instructions on file upload and text input

Reply:
Updated the instructions based on latest Chat elements.